independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/Qualifiers.java

```diff
@@ -23,15 +25,32 @@ private Qualifiers() {
     }
 
     static void verify(Iterable<Annotation> qualifiers) {
+        Map<Class<? extends Annotation>, Integer> timesQualifierWasSeen = new HashMap<>();
         for (Annotation qualifier : qualifiers) {
             verifyQualifier(qualifier.annotationType());
+            timesQualifierWasSeen.compute(qualifier.annotationType(), (k, v) -> (v == null) ? 1 : (v + 1));
         }
+        checkQualifiersForDuplicates(timesQualifierWasSeen);
     }
 
     static void verify(Annotation... qualifiers) {
+        Map<Class<? extends Annotation>, Integer> timesQualifierWasSeen = new HashMap<>();
         for (Annotation qualifier : qualifiers) {
             verifyQualifier(qualifier.annotationType());
+            timesQualifierWasSeen.compute(qualifier.annotationType(), (k, v) -> (v == null) ? 1 : (v + 1));
         }
+        checkQualifiersForDuplicates(timesQualifierWasSeen);
+    }
+
+    // in various cases, specification requires to check qualifiers for duplicates and throw IAE
+    private static void checkQualifiersForDuplicates(Map<Class<? extends Annotation>, Integer> timesQualifierSeen) {
+        timesQualifierSeen.forEach((aClass, timesSeen) -> {
```

Comment:
I think that there's still a general rule to avoid lambdas at runtime unless it really helps with readability or simplifies the code significantly and I don't this this a case of one or the other...

Reply:
You're right, will do.